### PR TITLE
change !logs regex to allow words before/after

### PR DESCRIPTION
**Pull request by quiddity-wp**

### CODE REVIEW

File: snerk.py [hunk 1]
```diff
@@ -501,8 +501,7 @@ def privmsg(self, sender, channel, msg):
             self.msg(channel, random.choice(sayings))
             return
 
-        elif (re.search(r'^\s*!\s*logs?\s*$', msg, re.I|re.U) or
-              re.search(r'^\s*logs?\s*!\s*$', msg, re.I|re.U)):
+        elif (re.search(r'(?<=\W)!logs?(?=\W)|(?<=\W)logs?!(?=\W)'ig, msg, re.I|re.U)
```
Comment (mzmcbride) [inline snerk.py]: I very much doubt this line works currently. If nothing else, the mandatory trailing colon is missing. I don't know what you were going for with the `'ig` part.

Maybe something like `elif re.search(r'^([ !?]*logs?|logs?[ !?]*)$', msg, re.I|re.U):` would work (untested)? I'm still not sure of your exact requirements.

Comment (quiddity-wp) [inline snerk.py]: I was trying to adapt https://regex101.com/r/cY6bP4/4 from matma, but I don't understand how the "ig" is meant to fit in... (Oh! Now I look afresh, I understand that the "ig" are covering how the 'test string' area gets parsed. I shall remove, and add the colon...)

File: snerk.py [hunk 2]
```diff
@@ -501,8 +501,7 @@ def privmsg(self, sender, channel, msg):
             self.msg(channel, random.choice(sayings))
             return
 
-        elif (re.search(r'^\s*!\s*logs?\s*$', msg, re.I|re.U) or
-              re.search(r'^\s*logs?\s*!\s*$', msg, re.I|re.U)):
+        elif (re.search(r'(?<=\W)!logs?(?=\W)|(?<=\W)logs?!(?=\W)', msg, re.I|re.U):
```
Comment (mzmcbride) [inline snerk.py]: Mismatched parentheses. :-(

Comment (legoktm) [inline snerk.py]: You can should remove the `(` before the `re.search`.

**Comment by mzmcbride** [comment]
Let's try it out!

**Comment by mzmcbride** [comment]
Fixed for realz in 615e6a97.

**Pull request merged**